### PR TITLE
Expose and fix logger kwargs 

### PR DESCRIPTION
**Pull request by FynnBe**
This PR fixes that the changes for loader_kwargs did not spread in all the needed places
and adds them to the training checkpoint

also adds init_data logging for WandbLogger

### CODE REVIEW

File: torch_em/trainer/wandb_logger.py [hunk 1]
```diff
@@ -84,6 +77,12 @@ def log_train(self, step, loss, lr, x, y, prediction, log_gradients=False):
 
     def log_validation(self, step, metric, loss, x, y, prediction):
         wandb.log({"validation/loss": loss, "validation/metric": metric}, step=step)
+        if loss < self.wand_run.summary["validation/loss"]:
+            self.wand_run.summary["validation/loss"] = loss
+
+        if metric > self.wand_run.summary["validation/metric"]:
```
Comment (constantinpape) [inline torch_em/trainer/wandb_logger.py:83]: Wrong sign?!

**Review by constantinpape — APPROVED** [review]
Looks good overall, but please fix the sign of the metric and the tests.

**Comment by constantinpape** [comment]
@FynnBe feel free to merge once you are done; as long as tests pass

**Comment by FynnBe** [comment]
all done here 👍

**Pull request merged**